components/sup/src/manager/mod.rs

```diff
@@ -362,19 +477,21 @@ impl Manager {
                 gateway_state: Arc::new(RwLock::new(gateway_state)),
             }),
             self_updater: self_updater,
-            updater: ServiceUpdater::new(server.clone()),
+            updater: Arc::new(Mutex::new(ServiceUpdater::new(server.clone()))),
             census_ring: CensusRing::new(sys.member_id.clone()),
             butterfly: server,
             launcher: launcher,
             peer_watcher: peer_watcher,
             spec_watcher: spec_watcher,
-            user_config_watcher: UserConfigWatcher::new(),
+            user_config_watcher: Arc::new(RwLock::new(UserConfigWatcher::new())),
             spec_dir: spec_dir,
             fs_cfg: Arc::new(fs_cfg),
             organization: cfg.organization,
             service_states: HashMap::new(),
             sys: Arc::new(sys),
             http_disable: cfg.http_disable,
+            busy_services: Arc::new(Mutex::new(HashSet::new())),
+            service_reconciliation_flag: ReconciliationFlag::default(),
```

Comment:
This should be addressed in https://github.com/habitat-sh/habitat/pull/6107/commits/710a73c9f9c13b533a375ed0aa93e2388d703d53

Reply:
Not that it should be a major sticking point or anything, but my feedback was about being able to understand what the state of `ReconciliationFlag` was by looking at the call-site and the naming of the function. To me `::new()` is no more informative than `::default()` since they both require looking at the implementation and/or comments, whereas `::from(true)` or `::true()` are explicitly clear and part of the code at the call-site.

Is there a reason we need `Default` on this type? It's nice to have when part of a larger type to derive `Default`, but `Manager` doesn't implement `Default`, does it?

Reply:
Gotcha... I'll have `new` take a boolean (`true` and `false` can't be function names, sadly).